### ISSUE TITLE
Beacon doesn't handle state restoration properly when navigating straight to contact form.

### ISSUE THREAD

**Issue by sipersso** [helpscout/beacon-android-sdk-sample#83]
* Beacon version: 2.0.2
* Android OS version: Android P
* Device make/model: Pixel 2

**Steps to reproduce**
1. Enable don't keep activities in developer settings on your phone
2. Start beacon with docs disabled and to the contact form. I am using the following code:
 ''' 
Beacon.setConfigOverrides(BeaconConfigOverrides(docsEnabled =false))
BeaconActivity.open(myActivity, BeaconScreens.CONTACT_FORM_SCREEN, ArrayList())
'''
3: Minimize the app 
4: Open the app again
..

**Actual outcome**
Application opens and only a spinner is shown

**Expected outcome**
The contact form should be shown.

**Comment by shaunmcilroy** [comment]
Hey @sipersso thanks for taking the time to log this for the team. As **Don't keep activities** is a developer option which most users in general won't have enabled this option we don't anticipate that this will exhibit to many customers. Thankfully this is the first report we've received for this issue. 

It is true though that this could indicate edge case behaviour where if the app hosting Beacon is killed and then the user returned to the app they may see this issue. We've got this logged up internally at the ticket below and will keep you updated on any developments regarding an improvement here:
https://helpscout.atlassian.net/browse/ANDSDK-352

**Comment by sipersso** [comment]
Hi! I only included don't keep activities since it is an easy way to detect state restoration errors. I did get this error on the app while field testing it without don't keep activities enabled. 

State restoration on Android can happen for many reasons. One common case is if the user is using a memory intensive app like YouTube or having a lot of applications open. While it is true that it isn't that common, it does happen every now and then ;)
